### PR TITLE
feat: NFT details new design

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

PR that updates the design for the NFT page.
Designs are here: https://www.figma.com/design/TfVzSMJA8KwpWX8TTWQ2iO/Asset-list-and-details?node-id=3717-47944&m=dev

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25524?quickstart=1)

## **Related issues**

Fixes:
Related: https://github.com/MetaMask/core/pull/4443

## **Manual testing steps**

1. Go to NFT tab
2. Click on any NFT you have
3. You should be able to see the new NFT page with no errors even if some data is not available.

### **Before**

https://github.com/MetaMask/metamask-extension/assets/10994169/dc6ca128-5f2a-45e2-9289-d84f5403d772

### **After**

https://github.com/MetaMask/metamask-extension/assets/10994169/58fd8cd4-24d9-42e9-931a-c33f8ae812d6

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .yarn/patches/@metamask-assets-controllers-npm-34.0.0-ea790e90a1.patch
Comment (sahar-fehri) [reply .yarn/patches/@metamask-assets-controllers-npm-34.0.0-ea790e90a1.patch:1]: Hey Pedro! I added a link to my core PR https://github.com/MetaMask/core/pull/4443 in the description;

I added the patch because i wanted to have one more approval on the core PR before merging. But i agree that ideally, upgrading assets-controllers would be better..

**Review by pedronfigueiredo — DISMISSED** [review]
LGTM, and it's great to see this cool new design shipping!

File: ui/components/app/nft-details/nft-detail-description.js
Comment (salimtb) [inline ui/components/app/nft-details/nft-detail-description.js:74]: this is better if you can mark value as required :
```
value: PropTypes.string.required,
```

File: ui/components/app/nft-details/nft-full-image.js
Comment (salimtb) [inline ui/components/app/nft-details/nft-full-image.js:52]:
```suggestion
  const [visible, setVisible] = useState(false);

  useEffect(() => {
    setVisible(true);
  }, []);
```
can be replaced with
```
const [visible, setVisible] = useState(true);
```

File: ui/components/app/nft-details/nft-full-image.js
Comment (salimtb) [inline ui/components/app/nft-details/nft-full-image.js:55]: you can use `<Box>` instead of div

File: ui/components/app/nft-options/nft-options.js
Comment (salimtb) [inline ui/components/app/nft-options/nft-options.js:18]: comment to remove here

File: ui/components/app/nft-details/nft-full-image.js
Comment (sahar-fehri) [reply ui/components/app/nft-details/nft-full-image.js:52]: Don't think this suggestion would work;
I need the visible to be initially false, and then after the component is rendered set it to true to have the transition effect.

File: ui/components/app/nft-details/nft-detail-description.js
Comment (sahar-fehri) [reply ui/components/app/nft-details/nft-detail-description.js:1]: Emm I think when i started created nft-details component in js, i finished writing the rest of the components in Js too,
Does this require to convert the rest of the components to tsx too?

Comment (sahar-fehri) [reply ui/components/app/nft-details/nft-detail-description.js]: I updated the new created components to `.tsx`; i kept the `nft-details.js` because when i wanted to update it to tsx and use `Nft` type from core, realized it would require also changing the send.js component.

**Comment by sahar-fehri** [comment]
@metamaskbot update-policies

File: ui/components/app/nft-details/nft-detail-description.js
Comment (sahar-fehri) [reply ui/components/app/nft-details/nft-detail-description.js]: Updated the files to .tsx 🙏

**Review by salimtb — DISMISSED** [review]

File: ui/helpers/utils/util.js
Comment (bergeron) [inline ui/helpers/utils/util.js:67]: I don't think this will localize, since I still see english months after switching languages.  Might want to try `Intl.DateTimeFormat`.  Similar to this snippet, but with different arguments so it's just the month/day/year.

https://github.com/MetaMask/metamask-extension/blob/7b3450a294a2fe5751726a9c33d6fa0b564f03dd/ui/pages/asset/util.ts#L3-L10

**Review by bergeron — DISMISSED** [review]
tested a bunch of my NFTs and looked great to me.  Left 1 datetime localization nitpick

File: ui/components/app/nft-details/nft-details.tsx
Comment (bergeron) [inline ui/components/app/nft-details/nft-details.tsx:364]: Does this work if the user configures a currency other than USD in MM settings?

`lastSale?.price?.amount?.usd` looks like it would always be USD, but then would this format it into euros without converting?

File: ui/components/app/nft-details/nft-details.tsx
Comment (bergeron) [inline ui/components/app/nft-details/nft-details.tsx:419]: `getPricePrecision` is probably not needed throughout this file, and the `precision` argument to `formatCurrency` could probably be omitted.

The default precision of `formatCurrency` uses the minimum of whatever the fiat currency is.  Like 2 for USD since $.01 is the minimum unit of a dollar.  I expect that's already correct for these values.  Higher precision was only needed for token prices like when 1 PEPE can be worth less than a penny like $0.00001082.

**Review by bergeron — DISMISSED** [review]

**Review by bergeron — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**
